packages/notifications/push/amplify_push_notifications/lib/amplify_push_notifications.dart

```diff
@@ -0,0 +1,236 @@
+    throw UnimplementedError(
+      'onNotificationReceivedInBackground() has not been implemented.',
+    );
+  }
+
+  @override
+  Future<void> configure({
+    AmplifyConfig? config,
+    required AmplifyAuthProviderRepository authProviderRepo,
+  }) async {
+    // Parse config values from amplifyconfiguration.json
+    final analyticsConfig = config?.analytics?.awsPlugin;
+    if (analyticsConfig == null) {
+      throw const AnalyticsException('No Pinpoint plugin config available');
+    }
+
+    if (_isConfigured) {
+      return;
+    }
+
+    // Block configure if registering device is not complete
+    // final deviceToken = await onTokenReceived.first;
+    // await _registerDevice(deviceToken);
+
+    // Initialize Endpoint Client
+    await _serviceProviderClient.init(
+      config: config,
+      authProviderRepo: authProviderRepo,
+    );
+
+    // Initialize listeners
+    onTokenReceived.listen(_tokenReceivedListener);
+    onNotificationReceivedInForeground.listen(_foregroundNotificationListener);
+    onNotificationOpened.listen(_notificationOpenedListener);
+
+    // TODO: Register the callback dispatcher
+
+    _logger.info('CONFIGURE API | Successfully configure push notifications');
+
+    _isConfigured = true;
+  }
+
+  void _foregroundNotificationListener(
+    PushNotificationMessage pushNotificationMessage,
+  ) {
+    _logger.info(
+      'Successfully listrening to foreground events: $pushNotificationMessage',
+    );
+
+    // TODO(Samaritan1011001): Record Analytics
+  }
+
+  void _backgroundNotificationListener(
+    PushNotificationMessage pushNotificationMessage,
+  ) {
+    // TODO(Samaritan1011001): Record Analytics
+  }
+
+  void _notificationOpenedListener(
+    PushNotificationMessage pushNotificationMessage,
+  ) {
+    _logger.info(
+      'Successfully listrening to notification opened events: $pushNotificationMessage',
+    );
+    // TODO(Samaritan1011001): Record Analytics
+  }
+
+  Future<String> _tokenReceivedListener(
+    String address,
+  ) async {
+    _logger.info('Successfully fetched the address: $address');
+    await _registerDevice(address);
+
+    return address;
+  }
+
+  Future<void> _registerDevice(String address) async {
+    try {
+      await _serviceProviderClient.registerDevice(address);
+      _logger.info('Successfully registered device with the servvice provider');
+      // return true;
+    } on Exception {
+      throw const PushNotificationException(
+        'Error when registering device with the service provider: ',
+      );
+    }
+  }
+
+  @override
+  Future<bool> requestPermissions({
+    bool? alert = true,
+    bool? badge = true,
+    bool? sound = true,
+  }) async {
+    return await _methodChannel.invokeMethod<bool>('requestPermissions', {
+          'alert': alert,
+          'badge': badge,
+          'sound': sound,
+        }) ??
+        false;
+  }
+
+  @override
+  PushNotificationMessage? get launchNotification => PushNotificationMessage();
```

Comment:
Can this be a getter?

By API definition, after calling `getLaunchNotification` the cached "launch notification" will be erased, so the subsequent call to `getLaunchNotification` should return `null`. With a getter this is not implementable correct?

Reply:
Why would it be erased?

Definitely possible, though:

```dart
PushNotificationMessage? _launchNotification;
PushNotificationMessage? get launchNotification {
  final launchNotification = _launchNotification;
  _launchNotification = null;
  return launchNotification;
}
```